dev_docs/tutorials/versioning_interfaces.mdx

```diff
@@ -0,0 +1,315 @@
+---
+id: kibDevTutorialVersioningInterfaces
+slug: /kibana-dev-docs/versioning-interfaces
+title: Versioning interfaces
+description: We need to keep old versions of interfaces available. This tutorial describes a strategy for managing versions of your interfaces over time.
+date: 2023-02-09
+tags: ['kibana', 'onboarding', 'dev', 'architecture']
+---
+
+To support versioned APIs we need to keep past versions of interfaces around, this tutorial presents one strategy to manage your interfaces.
+
+## The strategy
+
+Every plugin has a _domain_. A domain consists of a one or more concepts, usually objects, that are related in some logical way.
+
+At a high level the strategy for versioning our interfaces is:
+
+> Version a collection of related interfaces together. Whenever a single interface changes, increment the version of the entire collection of interfaces.
+
+Characteristics of this strategy:
+
+1. Avoid the `extends` keyword for referencing past interfaces. This results in cleaner API docs being generated.
+2. Leverage `* as` to create versioned namespaces rather than versioned interfaces.
+3. It is verbose, for a single change in a collection, we create a new version of the related collection.
```

Comment:
not 100% sure the intent of this sentence but I think it's this?

```suggestion
3. Although it is verbose, for a single change in a collection, we create a new version of the related collection.
```

Reply:
@lukeelmers What we're trying to get at is: "A new version of the related collection for every change to that collection in a verbose manner" or something along those lines.

ie. the 'verbosity' refers to the way versions are added to collections.

So yea, you're interpreting the sentence correctly.